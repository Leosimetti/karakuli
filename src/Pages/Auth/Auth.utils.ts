export type registerData = {
  email?: string
  username: string
  password: string
  passwordConfirm?: string
}

const URLmap = {
  login: 'https://karakuli-backend.herokuapp.com/api/v1/auth/jwt/login',
  register: 'https://karakuli-backend.herokuapp.com/api/v1/auth/register',
}

export async function submitHandle(data: registerData, mode: 'login' | 'register') {
  // console.log(data)
  const requestBody = getRequestBody(data, mode)
  const response = await fetch(URLmap[mode], {
    method: 'POST',
    mode: 'cors',
    body: requestBody,
    headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
  })
  const string = await response.text()

  console.log(JSON.parse(string))
}

function getRequestBody(data: registerData, mode: 'login' | 'register') {
  if (mode === 'login') {
    return `username=${data.email}&password=${data.password}`
  }

  return `email=${data.email}&username=${data.username}&password=${data.password}`
}
