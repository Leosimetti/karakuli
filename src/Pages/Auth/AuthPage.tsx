import React, { useState } from 'react';
import { FormProvider, useForm } from 'react-hook-form';
import { useDispatch } from 'react-redux';

import { AuthButton } from '../../Components/Button/AuthButton'
import { AuthNavButton } from '../../Components/Button/AuthNavButton'
import { FeatureCarousel } from '../../Components/Caroussel/Carousel'
import { AuthTextInput } from '../../Components/Input/AuthInput'
import { Logo } from '../../Components/Logo/Logo'
import { registerData, submitHandle } from '../../Data/api/actioncreators/register'

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import url from './layered-waves-haikei (1).svg'

import styled from 'styled-components'

const Page = styled.div`
  display: flex;
  height: 80vh;
  width: 70vw;
  flex-direction: row;
  justify-content: space-between;
  min-height: 540px;
  min-width: 600px;

  background: rgb(255, 210, 207);
  background: linear-gradient(172deg, rgba(255, 210, 207, 1) 0%, rgba(213, 80, 118, 1) 100%);
  padding: 10vh 15vw;

  @media (max-width: 1200px) {
    height: 85vh;
    width: 75vw;
    padding: 7.5vh 12.5vw;
  }

  @media (max-width: 992px) {
    height: 85vh;
    width: 85vw;
    padding: 7.5vh 7.5vw;
  }

  @media (max-width: 768px) {
    height: 100vh;
    width: 100vw;
    padding: 0;
  }

  @media (max-height: 820px) {
    height: 90vh;
    width: 90vw;
    padding: 5vh 5vw;
  }

  @media (max-height: 680px) {
    height: 100vh;
    width: 100vw;
    padding: 0;
  }
`;

const Column = styled.div<{ bgColor?: string; width: number }>`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex-direction: column;
  box-shadow: 15px 15px 10px rgba(169, 86, 86, 0.8);

  aspect-ratio: 600/900;
  height: 100%;
  width: ${(props) => props.width}%;

  background-color: ${(props) => props.bgColor || 'none'};
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  background-image: ${(props) => (props.bgColor ? 'none' : `url('${url}')`)};
  text-align: center;

  &:first-child {
    border-radius: 20px 0 0 20px;
  }

  &:last-child {
    border-radius: 0 20px 20px 0;
  }

  @media (max-width: 768px) {
    &:first-child,
    &:last-child {
      border-radius: 0;
    }
  }
  @media (max-height: 680px) {
    &:first-child,
    &:last-child {
      border-radius: 0;
    }
  }
`;

const Span = styled.span`
  font-family: Roboto, serif;
  font-style: normal;
  font-weight: normal;
  font-size: 28px;
  line-height: 32px;

  color: #000000;
  @media (max-width: 1200px) {
    font-size: 24px;
    line-height: 26px;
  }
  @media (max-width: 992px) {
    font-size: 20px;
    line-height: 20px;
  }
  @media (max-width: 768px) {
    font-size: 18px;
    line-height: 18px;
  }

  @media (max-height: 800px) {
    font-size: 20px;
    line-height: 20px;
  }
`;

const Blur = styled.div<{ align: string }>`
  width: 95%;
  background-color: rgb(135, 203, 252, 0.69);
  border-radius: 10px;
  text-align: ${(props) => props.align};

  //margin: 1% 0;
  padding: 15px 7px;
`;

const AuthNav = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 5%;
  width: 100%;
`;

const AuthContainer = styled.div`
  width: 80%;
  margin: 20px 0;
`;

export const AuthPage = () => {
  const [currentForm, setCurrentForm] = useState<'login' | 'register'>('login');
  const methods = useForm({
    mode: 'onSubmit',
    reValidateMode: 'onChange',
    defaultValues: {},
    resolver: undefined,
    context: undefined,
    criteriaMode: 'firstError',
    shouldFocusError: true,
    shouldUnregister: false,
    shouldUseNativeValidation: false,
    delayError: undefined,
  });
  const dispatch = useDispatch();

  async function onSubmit(data: registerData) {
    // console.log(data)
    dispatch(submitHandle(data, currentForm));
  }

  return (
    <Page>
      <Column width={60}>
        <Logo size="big" />
        <Blur align="left">
          <Span>
            <b>Karakuli</b>
            {' '}
            - платформа для изучения японского языка, применяющая технологию
            интервальных повторений для достижения быстрого результата
          </Span>
        </Blur>
        <Blur align="center">
          <Span>Доступен следующий функционал:</Span>
        </Blur>
        <FeatureCarousel />
      </Column>
      <Column bgColor="#303030" width={40}>
        <AuthContainer>
          <AuthNav>
            <AuthNavButton active={currentForm === 'login'} onClick={() => setCurrentForm('login')}>
              Login
            </AuthNavButton>
            <AuthNavButton
              active={currentForm === 'register'}
              onClick={() => setCurrentForm('register')}
            >
              Register
            </AuthNavButton>
          </AuthNav>
          <FormProvider {...methods}>
            <form onSubmit={methods.handleSubmit(onSubmit)}>
              <AuthTextInput
                title="Почта"
                icon={{
                  imageLink:
                    'https://emojipedia-us.s3.dualstack.us-west-1.amazonaws.com/thumbs/120/google/298/envelope_2709-fe0f.png',
                }}
                type="email"
                registerName="email"
              />

              {currentForm === 'register' && (
                <AuthTextInput
                  title="Логин"
                  icon={{
                    imageLink:
                      'https://emojipedia-us.s3.dualstack.us-west-1.amazonaws.com/thumbs/120/apple/285/bust-in-silhouette_1f464.png',
                  }}
                  registerName="username"
                />
              )}
              <AuthTextInput
                title="Пароль"
                icon={{
                  imageLink:
                    'https://emojipedia-us.s3.dualstack.us-west-1.amazonaws.com/thumbs/120/whatsapp/302/locked_1f512.png',
                }}
                type="password"
                registerName="password"
              />

              {currentForm === 'register' && (
                <AuthTextInput
                  title="Повторите пароль"
                  icon={{
                    imageLink:
                      'https://emojipedia-us.s3.dualstack.us-west-1.amazonaws.com/thumbs/120/whatsapp/302/locked_1f512.png',
                  }}
                  type="password"
                  registerName="passwordConfirm"
                  validate={(input) =>
                    // typescript error with documented utilization of library
                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
                    //  @ts-ignore
                    (input === methods.watch('password') ? true : 'Пароли должны совпадать')}
                />
              )}
              <AuthButton>{currentForm === 'register' ? 'Создать аккаунт' : 'Войти'}</AuthButton>
            </form>
          </FormProvider>
        </AuthContainer>
      </Column>
    </Page>
  );
};
