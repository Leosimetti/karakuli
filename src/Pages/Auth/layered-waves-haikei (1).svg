<svg id="visual" viewBox="0 0 600 900" width="600" height="900" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1"><rect x="0" y="0" width="600" height="900" fill="#ffffff"></rect><path d="M0 341L14.3 314.3C28.7 287.7 57.3 234.3 85.8 246.7C114.3 259 142.7 337 171.2 353.3C199.7 369.7 228.3 324.3 257 316.2C285.7 308 314.3 337 343 337.3C371.7 337.7 400.3 309.3 428.8 295.3C457.3 281.3 485.7 281.7 514.2 294.5C542.7 307.3 571.3 332.7 585.7 345.3L600 358L600 901L585.7 901C571.3 901 542.7 901 514.2 901C485.7 901 457.3 901 428.8 901C400.3 901 371.7 901 343 901C314.3 901 285.7 901 257 901C228.3 901 199.7 901 171.2 901C142.7 901 114.3 901 85.8 901C57.3 901 28.7 901 14.3 901L0 901Z" fill="#ffd2cf"></path><path d="M0 349L14.3 351.7C28.7 354.3 57.3 359.7 85.8 380.3C114.3 401 142.7 437 171.2 422.8C199.7 408.7 228.3 344.3 257 349.2C285.7 354 314.3 428 343 452.8C371.7 477.7 400.3 453.3 428.8 424.2C457.3 395 485.7 361 514.2 370.8C542.7 380.7 571.3 434.3 585.7 461.2L600 488L600 901L585.7 901C571.3 901 542.7 901 514.2 901C485.7 901 457.3 901 428.8 901C400.3 901 371.7 901 343 901C314.3 901 285.7 901 257 901C228.3 901 199.7 901 171.2 901C142.7 901 114.3 901 85.8 901C57.3 901 28.7 901 14.3 901L0 901Z" fill="#f7b3b4"></path><path d="M0 461L14.3 466.2C28.7 471.3 57.3 481.7 85.8 485.5C114.3 489.3 142.7 486.7 171.2 476.2C199.7 465.7 228.3 447.3 257 465.2C285.7 483 314.3 537 343 556.2C371.7 575.3 400.3 559.7 428.8 540.2C457.3 520.7 485.7 497.3 514.2 489.2C542.7 481 571.3 488 585.7 491.5L600 495L600 901L585.7 901C571.3 901 542.7 901 514.2 901C485.7 901 457.3 901 428.8 901C400.3 901 371.7 901 343 901C314.3 901 285.7 901 257 901C228.3 901 199.7 901 171.2 901C142.7 901 114.3 901 85.8 901C57.3 901 28.7 901 14.3 901L0 901Z" fill="#ed949c"></path><path d="M0 579L14.3 594.3C28.7 609.7 57.3 640.3 85.8 654.2C114.3 668 142.7 665 171.2 640C199.7 615 228.3 568 257 568.7C285.7 569.3 314.3 617.7 343 625.5C371.7 633.3 400.3 600.7 428.8 596.8C457.3 593 485.7 618 514.2 624.5C542.7 631 571.3 619 585.7 613L600 607L600 901L585.7 901C571.3 901 542.7 901 514.2 901C485.7 901 457.3 901 428.8 901C400.3 901 371.7 901 343 901C314.3 901 285.7 901 257 901C228.3 901 199.7 901 171.2 901C142.7 901 114.3 901 85.8 901C57.3 901 28.7 901 14.3 901L0 901Z" fill="#e27487"></path><path d="M0 670L14.3 671.2C28.7 672.3 57.3 674.7 85.8 682C114.3 689.3 142.7 701.7 171.2 693C199.7 684.3 228.3 654.7 257 655C285.7 655.3 314.3 685.7 343 703.7C371.7 721.7 400.3 727.3 428.8 727.3C457.3 727.3 485.7 721.7 514.2 710C542.7 698.3 571.3 680.7 585.7 671.8L600 663L600 901L585.7 901C571.3 901 542.7 901 514.2 901C485.7 901 457.3 901 428.8 901C400.3 901 371.7 901 343 901C314.3 901 285.7 901 257 901C228.3 901 199.7 901 171.2 901C142.7 901 114.3 901 85.8 901C57.3 901 28.7 901 14.3 901L0 901Z" fill="#d55076"></path><path d="M0 814L14.3 800C28.7 786 57.3 758 85.8 752.7C114.3 747.3 142.7 764.7 171.2 766.2C199.7 767.7 228.3 753.3 257 749.5C285.7 745.7 314.3 752.3 343 763.8C371.7 775.3 400.3 791.7 428.8 789.2C457.3 786.7 485.7 765.3 514.2 754.2C542.7 743 571.3 742 585.7 741.5L600 741L600 901L585.7 901C571.3 901 542.7 901 514.2 901C485.7 901 457.3 901 428.8 901C400.3 901 371.7 901 343 901C314.3 901 285.7 901 257 901C228.3 901 199.7 901 171.2 901C142.7 901 114.3 901 85.8 901C57.3 901 28.7 901 14.3 901L0 901Z" fill="#c62368"></path></svg>