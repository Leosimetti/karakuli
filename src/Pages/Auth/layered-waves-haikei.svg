<svg id="visual" viewBox="0 0 600 900" width="600" height="900" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1"><rect x="0" y="0" width="600" height="900" fill="#303030"></rect><path d="M0 361L14.3 348.8C28.7 336.7 57.3 312.3 85.8 306.8C114.3 301.3 142.7 314.7 171.2 323.2C199.7 331.7 228.3 335.3 257 335.7C285.7 336 314.3 333 343 325.3C371.7 317.7 400.3 305.3 428.8 294.5C457.3 283.7 485.7 274.3 514.2 278.7C542.7 283 571.3 301 585.7 310L600 319L600 0L585.7 0C571.3 0 542.7 0 514.2 0C485.7 0 457.3 0 428.8 0C400.3 0 371.7 0 343 0C314.3 0 285.7 0 257 0C228.3 0 199.7 0 171.2 0C142.7 0 114.3 0 85.8 0C57.3 0 28.7 0 14.3 0L0 0Z" fill="#fa7268"></path><path d="M0 300L14.3 293.5C28.7 287 57.3 274 85.8 259.8C114.3 245.7 142.7 230.3 171.2 222.7C199.7 215 228.3 215 257 219.3C285.7 223.7 314.3 232.3 343 236.7C371.7 241 400.3 241 428.8 250C457.3 259 485.7 277 514.2 271C542.7 265 571.3 235 585.7 220L600 205L600 0L585.7 0C571.3 0 542.7 0 514.2 0C485.7 0 457.3 0 428.8 0C400.3 0 371.7 0 343 0C314.3 0 285.7 0 257 0C228.3 0 199.7 0 171.2 0C142.7 0 114.3 0 85.8 0C57.3 0 28.7 0 14.3 0L0 0Z" fill="#ef5f67"></path><path d="M0 158L14.3 154C28.7 150 57.3 142 85.8 153.2C114.3 164.3 142.7 194.7 171.2 201C199.7 207.3 228.3 189.7 257 175.3C285.7 161 314.3 150 343 145.7C371.7 141.3 400.3 143.7 428.8 153C457.3 162.3 485.7 178.7 514.2 178C542.7 177.3 571.3 159.7 585.7 150.8L600 142L600 0L585.7 0C571.3 0 542.7 0 514.2 0C485.7 0 457.3 0 428.8 0C400.3 0 371.7 0 343 0C314.3 0 285.7 0 257 0C228.3 0 199.7 0 171.2 0C142.7 0 114.3 0 85.8 0C57.3 0 28.7 0 14.3 0L0 0Z" fill="#e34c67"></path><path d="M0 160L14.3 156.7C28.7 153.3 57.3 146.7 85.8 145.2C114.3 143.7 142.7 147.3 171.2 148.2C199.7 149 228.3 147 257 148.2C285.7 149.3 314.3 153.7 343 154.7C371.7 155.7 400.3 153.3 428.8 143.7C457.3 134 485.7 117 514.2 112.8C542.7 108.7 571.3 117.3 585.7 121.7L600 126L600 0L585.7 0C571.3 0 542.7 0 514.2 0C485.7 0 457.3 0 428.8 0C400.3 0 371.7 0 343 0C314.3 0 285.7 0 257 0C228.3 0 199.7 0 171.2 0C142.7 0 114.3 0 85.8 0C57.3 0 28.7 0 14.3 0L0 0Z" fill="#d53867"></path><path d="M0 41L14.3 49.3C28.7 57.7 57.3 74.3 85.8 78C114.3 81.7 142.7 72.3 171.2 73C199.7 73.7 228.3 84.3 257 84.8C285.7 85.3 314.3 75.7 343 71.3C371.7 67 400.3 68 428.8 69.8C457.3 71.7 485.7 74.3 514.2 79.5C542.7 84.7 571.3 92.3 585.7 96.2L600 100L600 0L585.7 0C571.3 0 542.7 0 514.2 0C485.7 0 457.3 0 428.8 0C400.3 0 371.7 0 343 0C314.3 0 285.7 0 257 0C228.3 0 199.7 0 171.2 0C142.7 0 114.3 0 85.8 0C57.3 0 28.7 0 14.3 0L0 0Z" fill="#c62368"></path></svg>