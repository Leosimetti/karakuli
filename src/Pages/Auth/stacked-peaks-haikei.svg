<svg id="visual" viewBox="0 0 600 900" width="600" height="900" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1"><path d="M0 127L67 136L133 154L200 280L267 262L333 208L400 181L467 343L533 136L600 244L600 0L533 0L467 0L400 0L333 0L267 0L200 0L133 0L67 0L0 0Z" fill="#303030"></path><path d="M0 370L67 406L133 505L200 487L267 415L333 478L400 487L467 550L533 388L600 433L600 242L533 134L467 341L400 179L333 206L267 260L200 278L133 152L67 134L0 125Z" fill="#4c3339"></path><path d="M0 505L67 550L133 640L200 595L267 577L333 667L400 604L467 622L533 604L600 568L600 431L533 386L467 548L400 485L333 476L267 413L200 485L133 503L67 404L0 368Z" fill="#673442"></path><path d="M0 640L67 667L133 703L200 739L267 703L333 721L400 721L467 694L533 703L600 730L600 566L533 602L467 620L400 602L333 665L267 575L200 593L133 638L67 548L0 503Z" fill="#83324c"></path><path d="M0 775L67 784L133 784L200 793L267 829L333 838L400 829L467 784L533 820L600 784L600 728L533 701L467 692L400 719L333 719L267 701L200 737L133 701L67 665L0 638Z" fill="#9f2c56"></path><path d="M0 901L67 901L133 901L200 901L267 901L333 901L400 901L467 901L533 901L600 901L600 782L533 818L467 782L400 827L333 836L267 827L200 791L133 782L67 782L0 773Z" fill="#bb1f60"></path></svg>