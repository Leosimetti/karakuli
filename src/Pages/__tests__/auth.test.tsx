import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'

import { describe, expect, it } from '@jest/globals'
import getTestStore from '../../__tests__/testStore'
import { AuthPage } from '../Auth/AuthPage'

import { render } from '@testing-library/react'

describe('Auth Page', () => {
  it('renders', () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={getTestStore()}>
          <AuthPage />
        </Provider>
      </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })
})
