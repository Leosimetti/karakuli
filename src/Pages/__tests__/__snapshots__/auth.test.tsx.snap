// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Page renders 1`] = `
.c17 {
  background-color: #606060;
  box-shadow: 0 4px 4px rgba(0,0,0,0.25);
  color: #ffffff;
  width: 100%;
  border: none;
  border-radius: 15px;
  margin: 10px 0;
  font-size: 26px;
  line-height: 60px;
  height: 60px;
  font-family: Roboto,serif;
  font-style: normal;
  font-weight: normal;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  cursor: pointer;
}

.c17:hover,
.c17:focus,
.c17:active {
  background-color: #9b9b9b;
}

.c17:active {
  color: #303030;
}

.c10 {
  background-color: #9B9B9B;
  box-shadow: 0 4px 4px rgba(0,0,0,0.25);
  color: #ffffff;
  border: none;
  border-radius: 15px;
  margin: 10px 0;
  width: 100%;
  height: 60px;
  font-family: Roboto,serif;
  font-style: normal;
  font-weight: normal;
  font-size: 26px;
  line-height: 26px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  cursor: pointer;
}

.c10:hover,
.c10:focus {
  background-color: #9b9b9b;
  color: white;
  border: 3px #777777 solid;
}

.c11 {
  background-color: #444444;
  box-shadow: 0 4px 4px rgba(0,0,0,0.25);
  color: #9B9B9B;
  border: none;
  border-radius: 15px;
  margin: 10px 0;
  width: 100%;
  height: 60px;
  font-family: Roboto,serif;
  font-style: normal;
  font-weight: normal;
  font-size: 26px;
  line-height: 26px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  cursor: pointer;
}

.c11:hover,
.c11:focus {
  background-color: #9b9b9b;
  color: white;
  border: 3px #777777 solid;
}

.c6 {
  width: 75%;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c6 img {
  height: 200px;
  width: 300px;
}

.c6 .carousel-dots {
  margin: 0;
}

.c12 {
  text-align: left;
  width: 100%;
  margin: 10px 0;
  position: relative;
}

.c13 {
  position: relative;
  left: 0;
  color: #9b9b9b;
  font-family: Scada;
  font-style: normal;
  font-weight: normal;
  font-size: 26px;
}

.c14 {
  border-bottom: 6px solid;
  border-color: #9b9b9b;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c16 {
  background-color: transparent;
  outline: none;
  border: none;
  font-family: Scada;
  font-style: normal;
  font-weight: normal;
  font-size: 20px;
  width: calc(100% - 40px);
  color: #9b9b9b;
  margin-bottom: 5px;
  margin-left: 5px;
}

.c16:focus {
  color: #cdcdcd;
}

.c16:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 50px #303030 inset;
  -webkit-text-fill-color: #9b9b9b;
}

.c16:-webkit-autofill::first-line {
  font-family: Scada;
  font-size: 20px;
}

.c15 {
  height: 32px;
  width: 32px;
  margin-bottom: 7px;
  margin-top: 2px;
}

.c2 {
  font-family: Roboto,serif;
  font-style: normal;
  font-weight: normal;
  font-size: 108px;
  white-space: nowrap;
  line-height: 108px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #f18c9c;
  text-shadow: 1px 1px 2px red,0 0 1em #f18c9c,0 0 0.2em #f18c9c;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 80vh;
  width: 70vw;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: 540px;
  min-width: 600px;
  background: rgb(255,210,207);
  background: linear-gradient(172deg,rgba(255,210,207,1) 0%,rgba(213,80,118,1) 100%);
  padding: 10vh 15vw;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: 15px 15px 10px rgba(169,86,86,0.8);
  aspect-ratio: 600/900;
  height: 100%;
  width: 60%;
  background-color: none;
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  background-image: url('[object Object]');
  text-align: center;
}

.c1:first-child {
  border-radius: 20px 0 0 20px;
}

.c1:last-child {
  border-radius: 0 20px 20px 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: 15px 15px 10px rgba(169,86,86,0.8);
  aspect-ratio: 600/900;
  height: 100%;
  width: 40%;
  background-color: #303030;
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  background-image: none;
  text-align: center;
}

.c7:first-child {
  border-radius: 20px 0 0 20px;
}

.c7:last-child {
  border-radius: 0 20px 20px 0;
}

.c4 {
  font-family: Roboto,serif;
  font-style: normal;
  font-weight: normal;
  font-size: 28px;
  line-height: 32px;
  color: #000000;
}

.c3 {
  width: 95%;
  background-color: rgb(135,203,252,0.69);
  border-radius: 10px;
  text-align: left;
  margin: 10px;
  padding: 15px 7px;
}

.c5 {
  width: 95%;
  background-color: rgb(135,203,252,0.69);
  border-radius: 10px;
  text-align: center;
  margin: 10px;
  padding: 15px 7px;
}

.c9 {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 5%;
  width: 100%;
}

.c8 {
  width: 80%;
  margin: 20px 0;
}

@media (max-width:1200px) {
  .c17 {
    font-size: 22px;
  }
}

@media (max-width:768px) {
  .c17 {
    font-size: 18px;
  }
}

@media (max-width:1200px) {
  .c10 {
    font-size: 20px;
    height: 50px;
  }
}

@media (max-height:800px) {
  .c10 {
    font-size: 20px;
    height: 50px;
  }
}

@media (max-width:1200px) {
  .c11 {
    font-size: 20px;
    height: 50px;
  }
}

@media (max-height:800px) {
  .c11 {
    font-size: 20px;
    height: 50px;
  }
}

@media (max-width:1200px) {
  .c13 {
    font-size: 18px;
  }
}

@media (max-height:800px) {
  .c13 {
    font-size: 18px;
  }
}

@media (max-width:1200px) {
  .c16 {
    font-size: 18px;
  }
}

@media (max-width:1200px) {
  .c2 {
    font-size: 80px;
    line-height: 85px;
  }
}

@media (max-width:992px) {
  .c2 {
    font-size: 60px;
    line-height: 65px;
  }
}

@media (max-width:768px) {
  .c2 {
    font-size: 40px;
    line-height: 45px;
  }
}

@media (max-height:800px) {
  .c2 {
    font-size: 60px;
    line-height: 65px;
  }
}

@media (max-width:1200px) {
  .c0 {
    height: 85vh;
    width: 75vw;
    padding: 7.5vh 12.5vw;
  }
}

@media (max-width:992px) {
  .c0 {
    height: 85vh;
    width: 85vw;
    padding: 7.5vh 7.5vw;
  }
}

@media (max-width:768px) {
  .c0 {
    height: 100vh;
    width: 100vw;
    padding: 0;
  }
}

@media (max-height:820px) {
  .c0 {
    height: 90vh;
    width: 90vw;
    padding: 5vh 5vw;
  }
}

@media (max-height:680px) {
  .c0 {
    height: 100vh;
    width: 100vw;
    padding: 0;
  }
}

@media (max-width:768px) {
  .c1:first-child,
  .c1:last-child {
    border-radius: 0;
  }
}

@media (max-height:680px) {
  .c1:first-child,
  .c1:last-child {
    border-radius: 0;
  }
}

@media (max-width:768px) {
  .c7:first-child,
  .c7:last-child {
    border-radius: 0;
  }
}

@media (max-height:680px) {
  .c7:first-child,
  .c7:last-child {
    border-radius: 0;
  }
}

@media (max-width:1200px) {
  .c4 {
    font-size: 24px;
    line-height: 26px;
  }
}

@media (max-width:992px) {
  .c4 {
    font-size: 20px;
    line-height: 20px;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 18px;
    line-height: 18px;
  }
}

@media (max-height:800px) {
  .c4 {
    font-size: 20px;
    line-height: 20px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      width="60"
    >
      <span
        class="c2"
      >
        カラクリ
      </span>
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          <b>
            Karakuli
          </b>
           - платформа для изучения японского языка, применяющая технологию интервальных повторений для достижения быстрого результата
        </span>
      </div>
      <div
        class="c5"
      >
        <span
          class="c4"
        >
          Доступен следующий функционал:
        </span>
      </div>
      <div
        class="c6"
      >
        <div
          class="carousel-initialized"
          style="padding: 0px;"
        >
          <button
            class="carousel-arrow block carousel-prev"
            data-role="none"
            type="button"
          >
             
            Previous
          </button>
          <div
            class="carousel-track"
            style="height: 0px;"
          >
            <div
              class="carousel-item active"
              data-carouselkey="0"
              style="width: 1024px; z-index: 1;"
            >
              <div
                class="carousel-row"
              >
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="chkn"
                    src="[object Object]"
                  />
                  <h3>
                    Study Lists
                  </h3>
                </div>
              </div>
            </div>
            <div
              class="carousel-item"
              data-carouselkey="1"
              style="width: 1024px; z-index: 1;"
            >
              <div
                class="carousel-row"
              >
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="chkn"
                    src="https://elitemedicalprep.com/wp-content/uploads/2020/06/Osmosis-Spaced-Repetition-Graph.png"
                  />
                  <h3>
                    SRS
                  </h3>
                </div>
              </div>
            </div>
            <div
              class="carousel-item"
              data-carouselkey="2"
              style="width: 1024px; z-index: 1;"
            >
              <div
                class="carousel-row"
              >
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="chkn"
                    src="[object Object]"
                  />
                  <h3>
                    Lessons
                  </h3>
                </div>
              </div>
            </div>
          </div>
          <button
            class="carousel-arrow block carousel-next"
            data-role="none"
            type="button"
          >
             
            Next
          </button>
        </div>
        <ul
          class="carousel-dots"
          style="display: block;"
        >
          <li
            class="carousel-dots-active carousel-dot-1"
          >
            <button
              type="button"
            >
              1
            </button>
          </li>
          <li
            class=" carousel-dot-2"
          >
            <button
              type="button"
            >
              2
            </button>
          </li>
          <li
            class=" carousel-dot-3"
          >
            <button
              type="button"
            >
              3
            </button>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="c7"
      width="40"
    >
      <div
        class="c8"
      >
        <div
          class="c9"
        >
          <button
            class="c10"
          >
            Login
          </button>
          <button
            class="c11"
          >
            Register
          </button>
        </div>
        <form>
          <div
            class="c12"
          >
            <label
              class="c13"
              for="Почта"
            >
              Почта
            </label>
            <div
              class="c14"
            >
              <img
                alt="Почта"
                class="c15"
                src="https://emojipedia-us.s3.dualstack.us-west-1.amazonaws.com/thumbs/120/google/298/envelope_2709-fe0f.png"
              />
              <input
                class="c16"
                id="Почта"
                name="email"
                type="email"
              />
            </div>
          </div>
          <div
            class="c12"
          >
            <label
              class="c13"
              for="Пароль"
            >
              Пароль
            </label>
            <div
              class="c14"
            >
              <img
                alt="Пароль"
                class="c15"
                src="https://emojipedia-us.s3.dualstack.us-west-1.amazonaws.com/thumbs/120/whatsapp/302/locked_1f512.png"
              />
              <input
                class="c16"
                id="Пароль"
                name="password"
                type="password"
              />
            </div>
          </div>
          <button
            class="c17"
            type="submit"
          >
            Войти
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
